pkg/cloudprovider/aws/instancetypes.go

```diff
@@ -146,16 +145,31 @@ func (p *InstanceTypeProvider) getAllInstanceTypes(ctx context.Context) ([]*ec2.
 }
 
 // filterFrom returns a filtered list of instance types based on the provided resource constraints
-func (p *InstanceTypeProvider) filterFrom(instanceTypes []*packing.Instance, constraints *cloudprovider.Constraints, zones []string) []*packing.Instance {
+func (p *InstanceTypeProvider) filterFrom(instanceTypes []*packing.Instance, constraints AWSConstraints, zones []string) []*packing.Instance {
 	filtered := []*packing.Instance{}
 	architecture := utils.NormalizeArchitecture(constraints.Architecture)
 
 	for _, instanceTypeInfo := range instanceTypes {
-		if (len(constraints.InstanceTypes) == 0 || functional.ContainsString(constraints.InstanceTypes, *instanceTypeInfo.InstanceType)) &&
-			(len(constraints.InstanceTypes) != 0 || p.isDefaultInstanceType(instanceTypeInfo)) &&
-			(architecture == nil || functional.ContainsString(aws.StringValueSlice(instanceTypeInfo.ProcessorInfo.SupportedArchitectures), *architecture)) &&
-			functional.ContainsString(aws.StringValueSlice(instanceTypeInfo.SupportedUsageClasses), "on-demand") &&
-			(len(zones) == 0 || len(functional.IntersectStringSlice(instanceTypeInfo.Zones, zones)) > 0) {
+		if functional.All(
```

Comment:
yeah... I kind of agree.  If we go back to the `if` we'd need to keep upping the cyclo-complexity check in golangci-lint. It's currently set at 10 and this extra conditional sent it over the threshold.

Reply:
another way to scale this might be to put these funcs as private methods on the InstanceTypeProvider type and go back to a regular `If` or define a func that takes in an arbitrary list of bools:

```
   if isInstanceTypeSupported(constraints.InstanceTypes, instanceTypeInfo) && 
      isArchitectureSupported(architecture, instanceTypeInfo) &&
      isCapacityTypeSupported(constraints.CapacityType, instanceTypeInfo) &&
     isZonesSupported(zones, instanceTypeInfo) {
     ...
     }
   
```

Reply:
Hmm... I'm not sure the cyclo complexity check makes sense then?

Reply:
I want to keep the check in place. I like the most recent approach brandon posted.